cli/src/main/java/com/box/l10n/mojito/cli/command/PushCommand.java

```diff
@@ -146,4 +170,37 @@ public void execute() throws CommandException {
 
         consoleWriter.fg(Ansi.Color.GREEN).newLine().a("Finished").println(2);
     }
+
+    /**
+     * Get fileMatch list contains git changes against git tree HEAD
+     * @param sourceFileMatches
+     * @return
+     * @throws CommandException
+     */
+    private ArrayList<FileMatch> getGitChangeFiles(ArrayList<FileMatch> sourceFileMatches) throws CommandException, GitAPIException {
+        logger.debug("filter out source files which do not contain diff changes.");
+        GitRepository gitRepository = new GitRepository();
+        gitRepository.init(commandDirectories.getSourceDirectoryPath().toString());
+        Git git = new Git(gitRepository.jgitRepository);
+        String gitPath = gitRepository.getDirectory().getPath().replace(".git", "");
+        Set<String> gitDiffFiles = new HashSet<>();
+
+        Pattern pattern = Pattern.compile(gitRegex == null ? "." : gitRegex);
```

Comment:
since you're testing for `gitRegex` later, I think this can be simplified 

`Pattern pattern = gitRegex == null ? null :  Pattern.compile(gitRegex);`

Reply:
change to 

`Pattern pattern = gitRegex == null ? null : Pattern.compile(gitRegex);`